recipes/ncurses/all/conanfile.py

```diff
@@ -41,90 +49,85 @@ class NCursesConan(ConanFile):
         "with_pcre2": False,
     }
 
-    generators = "pkg_config"
-    exports_sources = "patches/*"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     @property
-    def _with_ticlib(self):
-        if self.options.with_ticlib == "auto":
-            return self.settings.os != "Windows"
-        else:
-            return self.options.with_ticlib
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
 
-    @property
-    def _with_tinfo(self):
-        if self.options.with_tinfo == "auto":
-            return self.settings.os != "Windows"
-        else:
-            return self.options.with_tinfo
+    def export_sources(self):
+        export_conandata_patches(self)
+        copy(self, "*.cmake", src=self.recipe_folder, dst=self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        # Set the default value based on OS
+        self.options.with_ticlib = self.settings.os != "Windows"
```

Comment:
This is overriding whatever the user is setting these options as even if they are not set to auto, right?

Reply:
I see https://github.com/conan-io/conan-center-index/pull/20355#discussion_r1397324700 but not clear to me what the correct way would be

Reply:
This only overwrites the default "auto" value with a correct OS-specific default value. User-specified values for these options are not affected.

Believe me, this is approach is positively fantastic and I'm glad it works. Any alternatives approaches would need an insane amount of hacking around `self.options` and `self.info.options` being expected in different methods.